### ISSUE TITLE
GitRepo.get_toppath() broken in direct mode

### ISSUE THREAD

**Issue by mih** [datalad/datalad#1179]
Stumbled upon this due to a test failure in #1169 but the bug is present in master too.

````
/tmp/demo % git init
Initialized empty Git repository in /tmp/demo/.git/
/tmp/demo (git)-[master] % git annex init
init  ok
(recording state in git...)
/tmp/demo (git)-[master] % git annex direct
commit  
On branch master

Initial commit

nothing to commit
ok
direct  ok
/tmp/demo (git)-[annex/direct/master] % python -c "from datalad.support.gitrepo import GitRepo; print(GitRepo.get_toppath('.'))"

mih@meiner /tmp/demo (git)-[annex/direct/master] % git annex indirect
commit  ok
indirect  ok
ok
mih@meiner /tmp/demo (git)-[master] % python -c "from datalad.support.gitrepo import GitRepo; print(GitRepo.get_toppath('.'))"
.
````

**Comment by mih** [comment]
Looks like we need to reimplement get_toppath() without `rev-parse`.

**Comment by bpoldrack** [comment]
Not necessarily. Since it's currently a classmethod it just fails to know we are in direct mode, which is bound to an instance of course:

```
tmp/demo % git annex direct  
commit  
On branch master

Initial commit

nothing to commit
ok
direct  ok
/tmp/demo % git rev-parse --show-toplevel
/tmp/demo % git -c core.bare=false rev-parse --show-toplevel
/tmp/demo
ben@tree /tmp/demo % 
```

So, we could fix it by binding it to instances instead.

**Comment by mih** [comment]
Cool!

**Comment by mih** [comment]
@bpoldrack sadly, this doesnt seem to work with git 2.11.0

**Comment by bpoldrack** [comment]
Oh, okay. What does happen with 2.11.0? Just the same result as without the option?

**Comment by mih** [comment]
Setting core.bare via -c just for the call has no effect. But if I change it in the config file it does what you showed...

**Comment by bpoldrack** [comment]
That's interesting. May be an even more general issue with 2.11.0

**Comment by bpoldrack** [comment]
Correction: Actually, it doesn't make sense to bind it to Repo instances, since we wouldn't ask for toppath if there was an instance already. Instead I overwrote it within `AnnexRepo` (still as a classmethod).
For now, this means `GitRepo.get_toppath` fails in direct mode, while `AnnexRepo.get_toppath` doesn't. This will change when I can merge PR #1091, for this provides a dedicated place for things like that. ATM just use `AnnexRepo.get_toppath`.

However, note, that the issue with git 2.11.0 isn't resolved yet. I'll send a PR against Michael's and reference this issue.

**Comment by bpoldrack** [comment]
Another observation worth noting: According to the tests, `get_toppath()` works with annex V6.

**Comment by mih** [comment]
yes, they have a worktree.

**Comment by mih** [comment]
Two fixes would come with a merge of #1169

**Comment by mih** [comment]
Fix merged.
